lambdas/build-proven-user-identity-details/src/test/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandlerTest.java

```diff
@@ -103,27 +100,22 @@ void setUp() {
 
     @Test
     void shouldReceive200ResponseCodeProvenUserIdentityDetails() throws Exception {
-        when(mockUserIdentityService.isVcSuccessful(any(), any())).thenCallRealMethod();
```

Comment:
`thenCallRealMethod` has been unreliable in testing. It seems like the interference of mocking the VcHelper (required for the new logic where we have to mock the filterVCBasedOnProfileType method) affects the VcHelper calls by the userIdentityService as it calls the real method

Reply:
Makes sense - I think it's a bit of a smell anyway, having a partially mocked class.